server/odc-service/src/main/java/com/oceanbase/odc/service/db/schema/syncer/column/AbstractDBColumnSyncer.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.db.schema.syncer.column;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.Ordered;
+
+import com.oceanbase.odc.metadb.dbobject.DBColumnEntity;
+import com.oceanbase.odc.metadb.dbobject.DBColumnRepository;
+import com.oceanbase.odc.metadb.dbobject.DBObjectEntity;
+import com.oceanbase.odc.metadb.dbobject.DBObjectRepository;
+import com.oceanbase.odc.service.connection.database.model.Database;
+import com.oceanbase.odc.service.db.schema.syncer.DBSchemaSyncer;
+import com.oceanbase.tools.dbbrowser.schema.DBSchemaAccessor;
+
+import lombok.NonNull;
+
+/**
+ * @author gaoda.xy
+ * @date 2024/4/10 10:05
+ */
+public abstract class AbstractDBColumnSyncer implements DBSchemaSyncer {
+
+    @Autowired
+    protected DBObjectRepository dbObjectRepository;
+
+    @Autowired
+    protected DBColumnRepository dbColumnRepository;
+
+    @Override
+    public void sync(@NonNull DBSchemaAccessor accessor, @NonNull Database database) {
+        Map<String, Set<String>> latestObject2Columns = getLatestObjectToColumns(accessor, database);
+        Map<String, DBObjectEntity> existingObject2Entity =
+                dbObjectRepository.findByDatabaseIdAndType(database.getId(), getObjectType()).stream()
+                        .collect(Collectors.toMap(DBObjectEntity::getName, e -> e, (e1, e2) -> e1));
+        if (CollectionUtils.isEmpty(existingObject2Entity.entrySet())) {
+            return;
+        }
+        Set<Long> existingObjectIds =
+                existingObject2Entity.values().stream().map(DBObjectEntity::getId).collect(Collectors.toSet());
+        Map<Long, List<DBColumnEntity>> existingObjectId2ColumnEntities =
```

Comment:
there should be too much in expressions if there are too many ob objects~ we need to avoid sql execution failure caused by too long sql